Add HS Interior Overhauls

HyliosSykes opened the pull request.

HyliosSykes committed:
Adding some Rules for HS Mods

pStyl3 commented:
There are a couple of issues with the PR as it is right now.

1. It adds an empty line above `  # JK's Skyrim - Patches #`, which seems random for this PR.

2. For each new plugin entry in the masterlist we require an `url` in the form of
```yaml
    url:
      - link: 'https://www.nexusmods.com/skyrimspecialedition/mods/86865/'
        name: 'HSWilderness - Anise''s Cabin'
```
right under the `name` tag. It seems that all these plugins have their [own mod pages](https://www.nexusmods.com/skyrimspecialedition/users/88670758?tab=user+files), so all the plugin entries of this PR would need to stay, if we wanted to merge it as is. However ..

3. For repeating things like multiple & identical `load after` or `msg` we utilize [regular expressions](https://loot.github.io/docs/contributing/Regular-Expression-Regex.html) in order to reduce duplicate/superfluous information. Such a regex could look like this (though I haven't gone and alphabetically sorted everything within it correctly, yet):
```yaml
  - name: 'HS(DragonBridge|Markarth|PlayerHomes|Riverwood|Solitude|Whiterun|Wilderness|Winterhold) - (Four Shields Tavern|Silver-Blood Inn|Alvor and Sigrid''s House|Faendal''s House|Hod and Gerdur''s House|Riverwood Trader|Sleeping Giant Inn|Sven and Hilde''s House|Temple of the Divines|The Winking Skeever|Arcadia''s Cauldron|Jorrvaskr|The Drunken Huntsman|Anise''s Cabin|Mage''s College|Breezehome|Hjerim|Honeyside|Proudspire Manor|SeverinManor|Vlindrel Hall)\.esp'
    after:
      - 'Immersive Citizens - AI Overhaul.esp'
      - 'EnhancedLightsandFX.esp'
      - 'Lux.esp'
    msg:
      - type: say
        content: 'Auto Sort after lighting mods to avoid overriding the changes made to the cell, use compatibility patches to remove the rest of the lights and FX that would cause flickering'
```
That way your message and the `after`'s wouldn't need to be added 21 times, but only one time.

4. It is also likely that your message can be swapped with one or more of our existing message anchors, which can be found in the [Prelude](https://github.com/loot/prelude/blob/v0.18/prelude.yaml). Though I will let @sibir-ine comment on that as well as the `after` data of this PR.

HyliosSykes commented:
Thank you, I'll try to fix these accordingly, the regexp is especially useful. Do I have to cancel the pull request to provide a new commit?

pStyl3 commented:
Let's wait and see what sibir has to say regarding this PR first. Afterwards you can push to your fork as before, the PR get's updated accordingly.

HyliosSykes commented:
Alright, would something like that be acceptable instead?

```yaml
  - name: 'HSDragonBridge - Four Shields Tavern.esp'
    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/73017' ]
  - name: 'HSMarkarth - Silver-Blood Inn.esp'
    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/78778' ]
  - name: 'HSPlayerHomes - Breezehome.esp'
    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/74184' ]
  - name: 'HSPlayerHomes - Honeyside.esp'
    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/70968' ]
  - name: 'HSPlayerHomes - Proudspire Manor.esp'
    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/72313' ]
  - name: 'HSPlayerHomes - SeverinManor.esp'
    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/69855' ]
  - name: 'HSPlayerHomes - Vlindrel Hall.esp'
    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/83170' ]
  - name: 'HSRiverwood - Alvor and Sigrid''s House.esp'
    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/86088' ]
  - name: 'HSRiverwood - Faendal''s House.esp'
    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/83574' ]
  - name: 'HSRiverwood - Hod and Gerdur''s House.esp'
    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/74491' ]
  - name: 'HSRiverwood - Riverwood Trader.esp'
    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/86090' ]
  - name: 'HSRiverwood - Sleeping Giant Inn.esp'
    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/86092' ]
  - name: 'HSRiverwood - Sven and Hilde''s House.esp'
    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/86094' ]
  - name: 'HSSolitude - Temple of the Divines.esp'
    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/76056' ]
  - name: 'HSSolitude - The Winking Skeever.esp'
    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/82417' ]
  - name: 'HSWhiterun - Arcadia''s Cauldron.esp'
    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/75340' ]
  - name: 'HSWhiterun - Jorrvaskr.esp'
    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/79389' ]
  - name: 'HSWhiterun - The Drunken Huntsman.esp'
    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/74909' ]
  - name: 'HSWilderness - Anise''s Cabin.esp'
    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/86865' ]
  - name: 'HS(DragonBridge|Markarth|PlayerHomes|Riverwood|Solitude|Whiterun|Wilderness|Winterhold) - (Four Shields Tavern|Silver-Blood Inn|Alvor and Sigrid''s House|Faendal''s House|Hod and Gerdur''s House|Riverwood Trader|Sleeping Giant Inn|Sven and Hilde''s House|Temple of the Divines|The Winking Skeever|Arcadia''s Cauldron|Jorrvaskr|The Drunken Huntsman|Anise''s Cabin|Breezehome|Honeyside|Proudspire Manor|SeverinManor|Vlindrel Hall)\.esp'
    after:
      - 'Immersive Citizens - AI Overhaul.esp'
      - 'EnhancedLightsandFX.esp'
      - 'Lux.esp'
    msg:
      - <<: *compatNotes
        subs: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/72307/?tab=forum&topic_id=12707772' ]
```

That way if I understand correctly, if I need to add broad rules that would apply for all my mods I would do it in the regexped bit and if I have rules that need to be added to a single file I could add some 'after' tags to this individual entry? Sorry if I sound dumb, it's the first time I'm attempting to do this kind of thing.

sibir-ine commented:
> Alright, would something like that be acceptable instead?

Typically, we separate mods from different pages even if they have some metadata in common, but I can see how it'd be more useful to group mods from the same "series" together. Just keep the cluster together in the *Location - Overhauls* section (the player home plugins too, the *Locations - Player Homes* section is more for new player homes anyway) & separate it from *all* surrounding plugins with an extra newline (you only did so with the plugin above in both existing clusters). Also, URLs should have trailing forward slashes if possible, &, preferably but not necessarily, they'd have `name`s that match the name of the page they're from. LOOT can use it when displaying hyperlinks for them instead of a generic "Location." Please see pStyl3's example above. Besides those points & those pStyl3 has already raised, there are some other minor issues I can point out more easily with GitHub's PR review feature.

> That way if I understand correctly, if I need to add broad rules that would apply for all my mods I would do it in the regexped bit and if I have rules that need to be added to a single file I could add some 'after' tags to this individual entry?

Yes, that's correct, though regex plugins should only be used for plugins from the same page (or, in this case, from the same "series").

sibir-ine commented:
Just FYI, you can make these changes in this PR; you don't have to close it & make a new one. Just push your changes to the same branch.

HyliosSykes committed:
Merge remote-tracking branch 'upstream/v0.18'

HyliosSykes commented:
Okay, just wanted to have something clean

sibir-ine commented:
You don't need to worry too much about that, we usually squash PRs into a single commit when merging.

HyliosSykes committed:
Update masterlist.yaml

Adding Rules for HS Mods

HyliosSykes commented:
Okay last commit should be the one, miss-clicked on the second -__-

Added the names for the url, grouped everything in the location area and removed the part I added in player homes. I think I removed the extra whitespace I added by mistake in the first commit but I wasn't able to compare to the original file (not used to github quite yet), hopefully the future PR will be less chaotic.

sibir-ine commented on masterlist.yaml:
```diff
@@ -21842,6 +21927,7 @@ plugins:
         util: 'SSEEdit v4.0.3g'
       - crc: 0xC8478AEC
         util: 'SSEEdit v4.0.3g'
+
```
As pStyl3 pointed out, this is irrelevant to the PR & should be removed:

```suggestion
```

sibir-ine commented on masterlist.yaml:
```diff
@@ -21511,6 +21511,91 @@ plugins:
+         name: 'HS Wilderness - Anise''s Cabin'
+  - name: 'HS(DragonBridge|Markarth|PlayerHomes|Riverwood|Solitude|Whiterun|Wilderness|Winterhold) - (Four Shields Tavern|Silver-Blood Inn|Alvor and Sigrid''s House|Faendal''s House|Hod and Gerdur''s House|Riverwood Trader|Sleeping Giant Inn|Sven and Hilde''s House|Temple of the Divines|The Winking Skeever|Arcadia''s Cauldron|Jorrvaskr|The Drunken Huntsman|Anise''s Cabin|Breezehome|Honeyside|Prou ...
+    after:
+      - 'Immersive Citizens - AI Overhaul.esp'
+      - 'EnhancedLightsandFX.esp'
```
These should be sorted alphabetically:

```suggestion
      - 'EnhancedLightsandFX.esp'
      - 'Immersive Citizens - AI Overhaul.esp'
```

sibir-ine commented on masterlist.yaml:
```diff
@@ -21511,6 +21511,91 @@ plugins:
+         name: 'HS Whiterun - The Drunken Huntsman'
+  - name: HSWilderness - Anise''s Cabin.esp
+    url:
+      -  link: 'https://www.nexusmods.com/skyrimspecialedition/mods/86865'
```
```suggestion
      -  link: 'https://www.nexusmods.com/skyrimspecialedition/mods/86865/'
```

sibir-ine commented on masterlist.yaml:
```diff
@@ -21511,6 +21511,91 @@ plugins:
   - name: 'Holds - Falkreath.esp'
     after: [ 'Landscape Fixes For Grass Mods.esp' ]
 
+  - name: HSDragonBridge - Four Shields Tavern.esp
```
It'd also help to add a subsection header to help clarify why these mods are all clustered together. If you already have an official name for this series, you could use that. If not, you could use something like this:

```suggestion
  ### HS Interior Locations ###
  - name: HSDragonBridge - Four Shields Tavern.esp
```

HyliosSykes commented:
I cannot seem to be able to approve the changes you made in review, it says "This diff has recently been updated. Please and try again. ", I did refresh but it changed nothing, do I need to make a new commit?

sibir-ine commented:
Ah, I don't recall what causes that, but, yeah, you'll have to manually commit the requested changes.

HyliosSykes committed:
Update masterlist.yaml

HyliosSykes commented:
Hopefully this will do, thank you for your patience. :)

sibir-ine commented on masterlist.yaml:
```diff
@@ -21511,6 +21511,92 @@ plugins:
   - name: 'Holds - Falkreath.esp'
     after: [ 'Landscape Fixes For Grass Mods.esp' ]
 
+### HS Interior Overhauls ###
```
This just needs to be indented once:

```suggestion
  ### HS Interior Overhauls ###
```

HyliosSykes replied:
Oh come on... How am I this bad?

sibir-ine replied:
No worries, it's easy to miss with how GH displays suggestions.

HyliosSykes committed:
Update masterlist.yaml

sibir-ine reviewed (approved):
Looks good to me, thanks! We can merge it once @pStyl3 approves of it as well.

HyliosSykes commented:
Brilliant, thanks a lot for your time, future PR will be smoother, I promise :)

pStyl3 commented on masterlist.yaml:
```diff
@@ -21511,6 +21511,92 @@ plugins:
     after: [ 'Landscape Fixes For Grass Mods.esp' ]
 
+  ### HS Interior Overhauls ###
+  - name: HSDragonBridge - Four Shields Tavern.esp
```
Plugin names should be written as
```yaml
  - name: 'Example.esp'
```
So the `'` is missing before and after the plugin names.

pStyl3 commented on masterlist.yaml:
```diff
@@ -21511,6 +21511,92 @@ plugins:
+  ### HS Interior Overhauls ###
+  - name: HSDragonBridge - Four Shields Tavern.esp
+    url:
+      -  link: 'https://www.nexusmods.com/skyrimspecialedition/mods/73017/'
+         name: 'HS DragonBridge - Four Shields Tavern'
```
There are two empty spaces inbetween the `-` and `link`, where it only should be one empty space.
The line with the `name` of the link therefore has one empty space too much as well.

HyliosSykes committed:
Update masterlist.yaml

HyliosSykes commented:
You're absolutely right, pretty unforgiving syntax :D

pStyl3 commented:
Looks good to me now too. :)
